cdc/sink/producer/kafka/kafka.go

```diff
@@ -428,43 +445,81 @@ func validateMaxMessageBytesAndCreateTopic(admin kafka.ClusterAdminClient, topic
 	return nil
 }
 
-func getBrokerMessageMaxBytes(admin kafka.ClusterAdminClient) (int, error) {
+func validateMinInsyncReplicas(admin kafka.ClusterAdminClient,
+	topics map[string]sarama.TopicDetail, topic string, replicationFactor int) error {
+	info, exists := topics[topic]
+	if exists {
+		minInsyncReplicasStr, err := getTopicConfig(admin, info,
+			kafka.MinInsyncReplicasConfigName,
+			kafka.MinInsyncReplicasConfigName)
+		if err != nil {
+			return err
+		}
+		minInsyncReplicas, err := strconv.Atoi(minInsyncReplicasStr)
+		if err != nil {
+			return err
+		}
+
+		if replicationFactor < minInsyncReplicas {
+			msg := fmt.Sprintf("`replication-factor` cannot be smaller than the `%s` of topic",
+				kafka.MinInsyncReplicasConfigName)
+			log.Error(msg, zap.Int("replicationFactor", replicationFactor),
+				zap.Int("minInsyncReplicas", minInsyncReplicas))
+			return errors.New(msg)
+		}
+	}
```

Comment:
What about extracting a function, and reuse it in L472-L487
```golang
minInsyncReplicasStr, err := minInsyncReplicasConfigGetter(...)
minInsyncReplicas, err := strconv.Atoi(minInsyncReplicasStr)
if replicationFactor < minInsyncReplicas {
    //...
}
```

Reply:
Good idea!

Reply:
Good idea! Addressed!